airbyte-integrations/connectors/destination-oracle/src/main/java/io/airbyte/integrations/destination/oracle/OracleOperations.java

```diff
@@ -150,8 +152,9 @@ private static void insertRawRecordsInSingleQuery(String tableName,
         int i = 1;
         for (final AirbyteRecordMessage message : records) {
           // 1-indexed
+          final JsonNode formattedData = StandardNameTransformer.formatJsonPath(message.getData());
```

Comment:
this is modifying the data itself. Is there any way around this?

Reply:
agreed with sherif. the name transformer was intended to transform "names" (e.g. table names and column names). in general we want to avoid modifying the data if at all possible.

Reply:
I'm doing the same as in BigQuery destination (this change the json path to remove any non-supported char in the stream name not change the original data):
https://github.com/airbytehq/airbyte/blob/1954284f6f45a4629fdaeaa9cf4f48fdd0619656/airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryRecordConsumer.java#L124-L129

Reply:
I updated the comment in PR, sorry the confuction: Oracle 19 json_value doesnt allow ' or " in keys so it we need to clean ~some records~ the key name using a formatted function. Example: ```column`_'with\"_quotes``` becomes ```column___with\__quotes```.

Reply:
Example of `AIRBYTE_DATA` saved using this function. The original data is kept but to run normalization I need a "cleaned" field name.
```
'{"date":"2020-08-29T00:00:00Z","partition":{"DATA":[{"currency":"EUR"}],"double_array_data":[[{"id":"EUR"}]],"column___with__quotes":[{"currency":"EUR"}],"column`_\'with\\"_quotes":[{"currency":"EUR"}]},"id":4.2}'
```

Reply:
fair enough. this feels like a weird sacrifice we are making for dbt.

Reply:
This is something needed for the destination itself because the source is writing a json key not supported.